code/game/machinery/bots/cprbot.dm

```diff
@@ -0,0 +1,332 @@
+
+/obj/structure/machinery/bot/cprbot/proc/patient_in_range()
+	if (human == null)
+		return FALSE
+
+	var/mob/living/carbon/human/patient = human.resolve()
+	if (patient == null)
+		return FALSE
+
+	return get_dist(src, patient) == 0
+
+/obj/structure/machinery/bot/cprbot/proc/move_to_target()
+	var/mob/living/carbon/human/patient = human.resolve()
+	// If we cannot see them anymore then stop moving
+	if (!can_still_see_patient())
+		go_idle()
+		return
+
+	// It might not exist anymore or something
+	if (patient == null)
+		go_idle()
+		return
+
+	if (!patient_in_range())
+		// We are already moving
+		if (state == STATE_CPRBOT_MOVING)
+			return
+
+		state = STATE_CPRBOT_MOVING
+		walk_to(src, patient, 0, movement_delay)
+	else
+		walk_to(src, 0) // make sure we stop walking
+		state = STATE_CPRBOT_CPR
+		switch_to_faster_processing()
+		try_perform_cpr()
+
+/obj/structure/machinery/bot/cprbot/proc/perform_cpr(mob/living/carbon/human/target)
+	if (!cpr_ready)
+		return
+
+	currently_healing = TRUE
+
+	update_icon()
+
+	target.revive_grace_period += 4 SECONDS
+	target.visible_message(SPAN_NOTICE("<b>[src]</b> automatically performs <b>CPR</b> on <b>[target]</b>."))
+	target.balloon_alert_to_viewers("Performing CPR, stay clear!")
+	currently_healing = TRUE
+	playsound(loc, 'sound/CPRbot/CPR.ogg', 25, 1)
+	cpr_ready = FALSE
+	addtimer(VARSET_CALLBACK(src, cpr_ready, TRUE), 7 SECONDS)
+
+/obj/structure/machinery/bot/cprbot/proc/try_perform_cpr()
+	currently_healing = TRUE
+	// Resolve the weak reference to check if the target still exists
+	var/mob/living/carbon/human/target = human.resolve()
+
+	if (!patient_in_range())
+		go_idle()
+		switch_to_slower_processing()
+		return
+
+	// Check if the target is valid and still needs CPR
+	if ((target.stat != DEAD) || (target.stat == DEAD && !target.check_tod()))
+		go_idle()
+		switch_to_slower_processing()
+		return
+
+	perform_cpr(target)
+
+/obj/structure/machinery/bot/cprbot/proc/self_destruct(mob/living/carbon/human/user = null)
+	var/obj/item/cprbot_item/I = new /obj/item/cprbot_item(src.loc)
+
+	playsound(loc, 'sound/CPRbot/CPRbot_poweroff.ogg', 25, 1)
+
+	if (user)
+		if (!user.put_in_active_hand(I))
+			if (!user.put_in_inactive_hand(I))
+				I.forceMove(src.loc)
+	else
+		I.forceMove(src.loc)
+
+	qdel(src)
+
+/obj/structure/machinery/bot/cprbot/attack_hand(mob/user as mob)
+	if (..())
+		return TRUE
+
+	if(!skillcheck(user, SKILL_MEDICAL, SKILL_MEDICAL_MEDIC))
+		visible_message(SPAN_DANGER("<B>[user] fails to undeploy [src] </B>"))
+		return FALSE
+
+	SEND_SIGNAL(user, COMSIG_LIVING_ATTACKHAND_HUMAN, src)
+
+	if (user != src)
+		visible_message(SPAN_DANGER("<B>[user] begins to undeploy [src]!</B>"))
+	self_destruct(user)
+	return TRUE
+
+/obj/structure/machinery/bot/cprbot/update_icon()
+	. = ..()
+
+	switch(state)
+		if (STATE_CPRBOT_IDLE)
+			icon_state = "cprbot0"
+		if (STATE_CPRBOT_CPR)
+			icon_state = "cprbot_active"
+
+/obj/structure/machinery/bot/cprbot/explode()
+	src.on = 0
```

Comment:
```suggestion
	src.on = FALSE
```

Even though 0 and FALSE are the same, should be using boolean defines when applicable to make it clear `on` is a boolean and not a random number.